cmd/tk/prefix.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/go-clix/cli"
+)
+
+func prefixCmds(prefix string) []*cli.Command {
+	ext_subcommands := map[string]string{}
+	if path, ok := os.LookupEnv("PATH"); ok {
+		paths := strings.Split(path, ":")
+		for _, p := range paths {
+			s, err := os.Stat(p)
+			if err == nil && s.IsDir() {
+				files, err := ioutil.ReadDir(p)
+				if err != nil {
+					panic(err)
+				}
+				for _, file := range files {
+					if strings.HasPrefix(file.Name(), prefix) &&
+						!file.IsDir() &&
+						file.Mode().IsRegular() &&
+						file.Mode().Perm()&0111 != 0 {
+						ext_subcommands[file.Name()] = fmt.Sprintf("%s/%s", p, file.Name())
```

Comment:
I think it is more go-seque to flatten this section out. Swap your if conditions, use continue or return and have flatter less nested code.